### PR TITLE
Weekone

### PR DESCRIPTION
**Pull request by stephenbilham**
This is my poll request

### CODE REVIEW

File: js-intro.js [hunk 1]
```diff
@@ -68,14 +119,31 @@ var testString = "thisisateststring"
...
+  }
+  return newStr
+}
+console.log(noVowels(testString))
 
+OR
 
-
-// 4b. STRETCH: Update your function to return an error message if the variable is not a string. Copy and paste your function to add these changes.
+noVowels = (str) => {
```
Comment (stephenbilham) [inline js-intro.js]: This is the or for 4A. & I did figure it out....

var testString = 2

var noVowels = function(string) {
  if (typeof string !== "string") {
  console.log('error, not a string');
  return
}
  var vowels = ['a','e','i','o', 'u'];
  var newStr = "";
  for (var i=0; i < string.length; i++) {
    if (!vowels.includes(string[i])) {
      newStr += string[i]
    }
  }
  return newStr
}
console.log(noVowels(testString))

File: js-intro.js [hunk 2]
```diff
@@ -68,14 +119,31 @@ var testString = "thisisateststring"
...
+  return str.replace(/[a,e,i,o,u]/ig, "")
+}
+console.log(noVowels(testString))
 
 
+// 4b. STRETCH: Update your function to return an error message if the variable is not a string. Copy and paste your function to add these changes.
 
+didnt figure out.
```
Comment (stephenbilham) [inline js-intro.js:146]: var testString = 2

var noVowels = function(string) {
if (typeof string !== "string") {
  console.log('error, not a string');
  return
}
var vowels = ['a','e','i','o', 'u'];
var newStr = "";
for (var i=0; i < string.length; i++) {
if (!vowels.includes(string[i])) {
newStr += string[i]
}
}
return newStr
}
console.log(noVowels(testString))

File: js-intro.js [hunk 3]
```diff
@@ -68,14 +119,31 @@ var testString = "thisisateststring"
...
+console.log(noVowels(testString))
 
+OR
 
-
-// 4b. STRETCH: Update your function to return an error message if the variable is not a string. Copy and paste your function to add these changes.
+noVowels = (str) => {
+  return str.replace(/[a,e,i,o,u]/ig, "")
+}
+console.log(noVowels(testString))
```
Comment (stephenbilham) [inline js-intro.js:141]: this is the OR for 4a... ^^^^

Comment (sjproctor) [reply js-intro.js:141]: Nice work approaching this both ways - Regex is a cool tool, but it's important to know how to code "the long way" too.

File: js-intro.md [hunk 4]
```diff
@@ -5,49 +5,61 @@ Answer the following questions. First, without external resources. Challenge you
...
-
+  Your answer: Part1. A function is a key word that is used in javaScript to call something withing your scope to perform a task. You need to define your function and you need to set an argument if necessary. 
+  Part 2. You would use a function to one, have effiencent code. The ability to have the functions syntax to repeat or perfom a task is overall useful as a dev. 
 
+  Researched answer: Functions are one of the fundamental building blocks in JavaScript. A function is a JavaScript procedure—a set of statements that performs a task or calculates a value. To use a function, you must define it somewhere in the scope from which you wish to call it.
 
 2. List 5 examples of JavaScript array methods.
 
-  Your answer:
+  Your answer: arr.join(), arr.map(), arr.toUpperCase, arr.Math.max() ?, arr.splice
```
Comment (sjproctor) [inline js-intro.md:16]: toUpperCase is a string method

**Review by sjproctor — APPROVED** [review]
Fantastic work. Your code looks really good.

**Pull request merged**